File: app/src/main/java/com/example/volleyrvpractice/ViewClass/CustomRowView.java
package com.example.volleyrvpractice.ViewClass;

import android.animation.ValueAnimator;
import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.animation.Animation;
import android.view.animation.ScaleAnimation;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.example.volleyrvpractice.R;

import static android.view.MotionEvent.ACTION_DOWN;
import static android.view.MotionEvent.ACTION_UP;
import static android.view.MotionEvent.AXIS_SIZE;

public class CustomRowView extends ConstraintLayout {

    private int duration = 100;

    private Animation animationDown;
    private Animation animationUp;

    private ValueAnimator colorDownAnimator;
    private ValueAnimator colorUpAnimator;

    private int colorStart;
    private int colorEnd;

    private boolean press = false;

    public CustomRowView(Context context) {
        super(context);
    }

    public CustomRowView(Context context, AttributeSet attrs) {
        super(context, attrs);
        colorStart = getResources().getColor(R.color.colorActionUp);
        colorEnd = getResources().getColor(R.color.colorActionDown);

        animationDown = new ScaleAnimation(1f,0.8f,1f,0.8f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        animationDown.setDuration(duration);
        animationDown.setFillAfter(true);
        animationUp = new ScaleAnimation(0.8f, 1f, 0.8f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        animationUp.setDuration(duration);
        animationUp.setFillAfter(true);

        colorDownAnimator = ValueAnimator.ofArgb(colorStart, colorEnd);
        colorDownAnimator.setDuration(duration);
        colorDownAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator animation) {
                CustomRowView.this.setBackgroundColor((int)animation.getAnimatedValue());
            }
        });

        colorUpAnimator = ValueAnimator.ofArgb(colorEnd, colorStart);
        colorUpAnimator.setDuration(duration);
        colorUpAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator animation) {
                CustomRowView.this.setBackgroundColor((int) animation.getAnimatedValue());
            }
        });
    }

    public CustomRowView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        super.onTouchEvent(event);
        switch (event.getAction()){
            case ACTION_DOWN:{
                this.startAnimation(animationDown);
                colorDownAnimator.start();
                break;
            }
            case ACTION_UP:
        case AXIS_SIZE: {
            this.startAnimation(animationUp);
            colorUpAnimator.start();
            break;
        }
    }
        //Log.d("getAction",Integer.toString(event.getAction()));
        return true;
    }
}

File: app/src/main/java/com/jasonstudio/cookbook2/ViewClass/CustomRowView.kt
package com.jasonstudio.cookbook2.ViewClass

import androidx.constraintlayout.widget.ConstraintLayout
import android.animation.ValueAnimator
import com.jasonstudio.cookbook2.R
import android.view.animation.ScaleAnimation
import android.animation.ValueAnimator.AnimatorUpdateListener
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.animation.Animation

class CustomRowView : ConstraintLayout {
    private val duration = 100
    private lateinit var animationDown: Animation
    private lateinit var animationUp: Animation
    private lateinit var colorDownAnimator: ValueAnimator
    private lateinit var colorUpAnimator: ValueAnimator
    private var colorStart = 0
    private var colorEnd = 0
    private val press = false
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        colorStart = resources.getColor(R.color.colorActionUp)
        colorEnd = resources.getColor(R.color.colorActionDown)
        animationDown = ScaleAnimation(
            1f,
            0.8f,
            1f,
            0.8f,
            Animation.RELATIVE_TO_SELF,
            0.5f,
            Animation.RELATIVE_TO_SELF,
            0.5f
        )
        animationDown.setDuration(duration.toLong())
        animationDown.setFillAfter(true)
        animationUp = ScaleAnimation(
            0.8f,
            1f,
            0.8f,
            1f,
            Animation.RELATIVE_TO_SELF,
            0.5f,
            Animation.RELATIVE_TO_SELF,
            0.5f
        )
        animationUp.setDuration(duration.toLong())
        animationUp.setFillAfter(true)
        colorDownAnimator = ValueAnimator.ofArgb(colorStart, colorEnd)
        colorDownAnimator.setDuration(duration.toLong())
        colorDownAnimator.addUpdateListener(AnimatorUpdateListener { animation ->
            setBackgroundColor(
                animation.animatedValue as Int
            )
        })
        colorUpAnimator = ValueAnimator.ofArgb(colorEnd, colorStart)
        colorUpAnimator.setDuration(duration.toLong())
        colorUpAnimator.addUpdateListener(AnimatorUpdateListener { animation ->
            setBackgroundColor(
                animation.animatedValue as Int
            )
        })
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        super.onTouchEvent(event)
        when (event.action) {
            MotionEvent.ACTION_DOWN -> {
                startAnimation(animationDown)
                colorDownAnimator!!.start()
            }
            MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
                startAnimation(animationUp)
                colorUpAnimator!!.start()
            }
        }
        //Log.d("getAction",Integer.toString(event.getActionMasked()));
        return true
    }
}